File: Cpp/1423_maximum_points_you_can_obtain_from_cards.cpp
// https://leetcode.cn/problems/maximum-points-you-can-obtain-from-cards/

class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int window = 0, total = 0;
        int window_len = n-k;
        for (int i = 0; i < window_len; ++i) {
            window += cardPoints[i];
            total += cardPoints[i];
        }
        int window_min = window;
        for (int i = window_len; i < n; ++i) {
            window += cardPoints[i] - cardPoints[i-window_len];
            window_min = min(window, window_min);
            total += cardPoints[i];
        }
        return total - window_min;
    }
};

File: Python3/1423_maximum_points_you_can_obtain_from_cards.py
# https://leetcode.cn/problems/maximum-points-you-can-obtain-from-cards/

class Solution:
    def maxScore(self, cardPoints, k):
        """
        Inputs:
            cardPoints: list[int]
            k: int
        Outputs:
            res: int
        """
        n = len(cardPoints)
        window, total = 0, 0
        window_len = n-k
        for i in range(window_len):
            window += cardPoints[i]
            total += cardPoints[i]
        window_min = window
        for i in range(window_len, n):
            window += (cardPoints[i] - cardPoints[i-window_len])
            window_min = min(window_min, window)
            total += cardPoints[i]
        return total - window_min